zigen-project/zen-remote
# Gl buffer

## Context

## Summary

  - minimum self-contained header file
  - move constructor to .cc
  - stop using smart pointer for non owning objects
- [x] add `gl-buffer`: representing the object created with `glCreateBuffers`.
- [x] add `virtual-object`
- [x] enable to commit state and use the state from rendering thread.

## How to check behavior

1. Build and install zen-remote
2. Build https://github.com/zigen-project/zen/pull/193
3. Start zen-desktop
4. Start `test-client` of zen-remote
5. Switch to immersive display system
6. Plug and unplug your pointing device
You will see the following log in the `test-client`

```log
== current resources ==
[remote] D:Sent UPD broadcast packet for client discovery
[remote] I:Server Found: 192.168.0.5
== current resources ==
Virtual Object:
  RenderingUnit:
== current resources ==
Virtual Object:
  RenderingUnit:
== current resources ==
== current resources ==
== current resources ==
== current resources ==
== current resources ==
Virtual Object:
  RenderingUnit:
== current resources ==
Virtual Object:
  RenderingUnit:
```
